### PR TITLE
added demo user

### PR DESCRIPTION
**Pull request by MillanAir**
### What this PR does (required):
- Added Protected Route functionality and Demo User Login

### Screenshots / Videos (required):
image

Demo User:
image

### Any information needed to test this feature (required):
- If user wants to browse a protected page then they need to log in first.

### Any issues with the current functionality (optional):
- Demo User login details aren't working fully. Needs extra information regarding the flow of login

### CODE REVIEW

File: client/src/pages/Login/LoginForm/LoginForm.tsx [hunk 1]
```diff
@@ -28,6 +29,27 @@ interface Props {
 
 export default function Login({ handleSubmit }: Props): JSX.Element {
   const classes = useStyles();
+  const history = useHistory();
+
+  (handleSubmit = (e) => {
```
Comment (rajivtitus) [inline client/src/pages/Login/LoginForm/LoginForm.tsx:34]: 1. Please add another button the Login Form called 'Demo Login'
2. Once completed, create a function to handle button click and use the `login()` function in the helpers folder and pass the demo user credentials as arguments.
3. Finally, reference the function in the button's onClick property :)

**Review by rajivtitus — CHANGES_REQUESTED** [review]

File: client/src/pages/Login/LoginForm/LoginForm.tsx (hunk 1 above)
Comment (MillanAir) [reply client/src/pages/Login/LoginForm/LoginForm.tsx]: Thanks for the support Rajiv, I added the functionality in login Form as per your steps mentioned. Thanks.

File: client/src/pages/Login/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -28,6 +31,27 @@ interface Props {
...
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        updateLoginContext(data.success);
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
+        console.error({ data });
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+    alert('Demo Login Works');
```
Comment (rajivtitus) [inline client/src/pages/Login/LoginForm/LoginForm.tsx:53]: You can remove this alert as the snack bar will be updated if there's an error and if there are no errors the user will be redirected to the dashboard.

**Review by rajivtitus — APPROVED** [review]
Looks good to me!

File: client/src/pages/Login/LoginForm/LoginForm.tsx (hunk 2 above)
Comment (MillanAir) [reply client/src/pages/Login/LoginForm/LoginForm.tsx:53]: ok sure

**Pull request merged**